### PR TITLE
Performance: Adds compiler optimize option

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Performance: Adds includes compiler optimize option

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj [hunk 1]
```diff
@@ -39,6 +39,7 @@
     <IncludeSource>false</IncludeSource>
     <RootNamespace>Microsoft.Azure.Cosmos</RootNamespace>
     <NoWarn>NU5125</NoWarn>
+    <Optimize>true</Optimize>
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:42]: Is there any benchmark showing the improvements? Does this also need to be done to the direct package?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:42]: Direct has it. Its a good to have.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:42]: This should be done on Release, not Debug, right?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:42]: For direct pacakge its unconditional. We can revisit if needed later.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj]: Good point @ealsur, done marked it for release only. 
Debug build debugging is impacted with-out it.

**Pull request closed without merging**